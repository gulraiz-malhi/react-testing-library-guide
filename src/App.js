import React from 'react';
import './App.css';
import TestAsync from './components/TestAsync';
import TestAxios from './components/TestAxios';
import CounterProvider from './components/TestContext';
import TestElements from './components/TestElements';
import TestEvents from './components/TestEvents';
import { FetchGreeting } from './components/TestNetworkAsync';
import TestRedux from './components/TestRedux';
import TestRouter from './components/TestRouter';

function App() {
  return (
    <div className="App">
      <h1 data-testid="custom-element">Testing 1</h1>
      <form>
        <label htmlFor="example">Example</label>
        <input placeholder="username" id="example" />
      </form>

      {/* <TestElements /> */}
       {/* <TestEvents /> */}
       {/* <TestAsync /> */}
      {/* <TestAxios url={'https://reqres.in/api/users?page=2'}/> */}
      {/* <TestRedux /> */}
      {/* <CounterProvider /> */}
      {/* <TestRouter /> */}
      {/* <FetchGreeting /> */}
    </div>
  );
}


export default App;
