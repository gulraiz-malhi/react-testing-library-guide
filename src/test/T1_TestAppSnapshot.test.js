import React from "react";
import { render, cleanup, screen } from "@testing-library/react";
import App from "../App";

describe("Take a snapshot", () => {
  afterEach(cleanup);
  it("should take a snapshot", () => {
    const { asFragment } = render(<App />);

    expect(asFragment(<App />)).toMatchSnapshot();
  });
});

describe("Test App", () => {
  it("renders ok", () => {
    render(<App />);
    expect(screen.getByText("Testing 1")).toBeDefined();
    expect(screen.getByLabelText(/example/i)).toBeDefined();
    expect(screen.getByPlaceholderText('username')).toBeDefined();
    expect(screen.getByTestId('custom-element')).toBeDefined();
  });
});
