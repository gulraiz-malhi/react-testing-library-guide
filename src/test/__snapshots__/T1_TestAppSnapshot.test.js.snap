// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Take a snapshot should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <h1
      data-testid="custom-element"
    >
      Testing 1
    </h1>
    <form>
      <label
        for="example"
      >
        Example
      </label>
      <input
        id="example"
        placeholder="username"
      />
    </form>
  </div>
</DocumentFragment>
`;
